### PR TITLE
storm vars, digraph edges, http tweaks

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/lib/node.py [hunk 1]
```diff
@@ -424,3 +430,19 @@ def delete(self, force=False):
         self.snap.buidcache.pop(self.buid)
 
         self.form.wasDeleted(self)
+
+class Path:
```
Comment (vEpiphyte) [inline synapse/lib/node.py]: What does this have to do with nodes?

Comment (invisig0th) [reply synapse/lib/node.py]: This tracks the progress of nodes passing through the storm runtime ( and can, for example, be used to record per-node variables and record pivot paths traversed to produce a given node )

File: synapse/datamodel.py [hunk 2]
```diff
@@ -357,6 +366,22 @@ def __init__(self):
...
+
+        info = {'doc': 'An digraph edge base type.'}
+        item = s_types.Edge(self, 'edge', info, {})
+        self.addBaseType(item)
+
+        info = {'doc': 'An digraph edge base type with a unique time.'}
+        item = s_types.TimeEdge(self, 'timeedge', info, {})
+        self.addBaseType(item)
+
+        info = {'doc': 'Arbitrary msgpack/json compatible data stored without an index.'}
```
[the remark this answers is not in the document]
Comment (invisig0th) [reply synapse/datamodel.py]: Changing to "msgpack"

File: synapse/lib/types.py [hunk 3]
```diff
@@ -646,7 +678,7 @@ def _normPyTuple(self, valu):
         try:
             formname, formvalu = valu
         except Exception as e:
-            raise s_exc.BadTypeValu(name=self.name, valu=valu)
+            raise s_exc.BadTypeValu(name=self.name, valu=valu, mesg=str(e))
```
[the remark this answers is not in the document]
Comment (invisig0th) [reply synapse/lib/types.py]: :D

File: synapse/lib/time.py [hunk 4]
```diff
@@ -74,3 +75,64 @@ def ival(*times):
         maxv += 1
 
     return (minv, maxv)
+
+onesec  = 1000
```
Comment (vEpiphyte) [inline synapse/lib/time.py]: These are duplicates of content in `synapse.lib.const`

File: synapse/tests/test_cortex.py [hunk 5]
```diff
@@ -9,6 +9,47 @@
...
 class CortexTest(s_test.SynTest):
 
+    def test_cortex_iter_props(self):
+
+        with self.getTestCore() as core:
+
+            with core.snap() as snap:
+
+                props = {'asn': 10, '.seen': '2016'}
+                node = snap.addNode('inet:ipv4', 0x01020304, props=props)
```
Comment (vEpiphyte) [inline synapse/tests/test_cortex.py]: use test forms, not model forms.

File: synapse/tests/test_cortex.py [hunk 6]
```diff
@@ -9,6 +9,47 @@
...
+                self.eq(node.get('asn'), 10)
+
+                props = {'asn': 20, '.seen': '2015'}
+                node = snap.addNode('inet:ipv4', 0x05050505, props=props)
+                self.eq(node.get('asn'), 20)
+
```
Comment (vEpiphyte) [inline synapse/tests/test_cortex.py]: use test forms, not model forms.

File: synapse/tests/test_cortex.py [hunk 7]
```diff
@@ -9,6 +9,47 @@
...
+        with self.getTestCore() as core:
+
+            with core.snap() as snap:
+
+                props = {'asn': 10, '.seen': '2016'}
+                node = snap.addNode('inet:ipv4', 0x01020304, props=props)
+                self.eq(node.get('asn'), 10)
+
+                props = {'asn': 20, '.seen': '2015'}
+                node = snap.addNode('inet:ipv4', 0x05050505, props=props)
```
Comment (vEpiphyte) [inline synapse/tests/test_cortex.py]: use test forms, not model forms.

File: synapse/models/inet.py [hunk 8]
```diff
@@ -225,8 +225,47 @@ def postTypeInit(self):
...
+
+            minv, maxv = self.getNetRange(text)
+
+            while minv < maxv:
+                yield minv
+                minv += 1
+
+            return
+
+        yield text
```
[the remark this answers is not in the document]
Comment (invisig0th) [reply synapse/models/inet.py]: That's ok, they pass through norm via addNode anyway
